### PR TITLE
perlPackages.FileMap and dep

### PR DESCRIPTION
**Pull request by invokesus**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -7205,6 +7205,21 @@ let
...
   };
 
+  FileMap = buildPerlModule rec {
+    pname = "File-Map";
+    version = "0.66";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/L/LE/LEONT/${pname}-${version}.tar.gz";
+      sha256 = "e73baecb5c0de715d24df68d56f74907977b77c951f8cc7e944ccb433620a589";
+    };
+    buildInputs = [ ModuleBuild TestFatal TestWarnings perl ];
```
Comment (volth) [inline pkgs/top-level/perl-packages.nix:7215]:
```suggestion
    buildInputs = [ TestFatal TestWarnings ];
```

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -14819,6 +14834,20 @@ let
...
   };
 
+  PerlIOLayers = buildPerlModule rec {
+    pname = "PerlIO-Layers";
+    version = "0.011";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/L/LE/LEONT/${pname}-${version}.tar.gz";
+      sha256 = "1d77e164aa5d5d24699a5479a423abdc1b07973fbe5a22d304fef2c560254b42";
+    };
+    buildInputs = [ ModuleBuild ];
```
Comment (volth) [inline pkgs/top-level/perl-packages.nix:14844]:
```suggestion

```

**Comment by invokesus** [comment]
Thanks for the review! Amended and force-pushed.

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -14819,6 +14834,19 @@ let
     };
   };
 
+  PerlIOLayers = buildPerlModule rec {
+    pname = "PerlIO-Layers";
+    version = "0.011";
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:14839]:
```suggestion
    version = "0.012";
```

File: pkgs/top-level/perl-packages.nix [hunk 4]
```diff
@@ -14819,6 +14834,19 @@ let
...
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/L/LE/LEONT/${pname}-${version}.tar.gz";
+      sha256 = "1d77e164aa5d5d24699a5479a423abdc1b07973fbe5a22d304fef2c560254b42";
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:14842]:
```suggestion
      sha256 = "1psaq3kwlk7g9rxvgsacfjk2mh6cscqf4xl7ggfkzfrnz91aabal";
```

**Review by stigtsp — APPROVED** [review]
Update PerlIO::Layers to 0.012

LGTM

File: pkgs/top-level/perl-packages.nix [hunk 5]
```diff
@@ -7205,6 +7205,21 @@ let
     propagatedBuildInputs = [ HTTPDate ];
   };
 
+  FileMap = buildPerlModule rec {
+    pname = "File-Map";
+    version = "0.66";
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:7210]:
```suggestion
    version = "0.67";
```

File: pkgs/top-level/perl-packages.nix [hunk 6]
```diff
@@ -7205,6 +7205,21 @@ let
...
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/L/LE/LEONT/${pname}-${version}.tar.gz";
+      sha256 = "e73baecb5c0de715d24df68d56f74907977b77c951f8cc7e944ccb433620a589";
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:7213]:
```suggestion
      sha256 = "1hpv4aprgypjxjx1kzbjnf6r29a98rw7mndlinixzk62vyz5sy0j";
```

**Review by stigtsp — COMMENTED** [review]
Update `File::Map` to 0.67

**Comment by invokesus** [comment]
Thanks for the review! Updated the versions and force-pushed.

**Comment by invokesus** [comment]
Oh, I see it got added in https://github.com/NixOS/nixpkgs/commit/4e8a420f545e99253b05a85a30551d4cdbbb261a. I'll close this then.

**Pull request closed without merging**